index.js

```diff
@@ -554,7 +554,11 @@ const plugin = fp(async function (app, opts) {
     if (shouldCompileJit) {
       if (!modifiedSchema && !modifiedDocument) {
         // can compile only when the schema and document are not modified
-        cached.jit = compileQuery(fastifyGraphQl.schema, document, operationName)
+        if (opts.compilerOptions) {
+          cached.jit = compileQuery(fastifyGraphQl.schema, document, operationName, opts.compilerOptions)
+        } else {
+          cached.jit = compileQuery(fastifyGraphQl.schema, document, operationName)
+        }
```

Comment:
how is this different from:

```suggestion
        cached.jit = compileQuery(fastifyGraphQl.schema, document, operationName, opts.compilerOptions)
```

?

Reply:
you are right! Simplified version is much better